### DISCUSSION TITLE
[AC] [Spell/Core]I find that 90% of the problems come from three reasons

### DISCUSSION THREAD

**Discussion by zbhcn** [trickerer/Trinity-Bots#320] [General]
I have been using your code for two months. Your bots are great.My friends and I have tested all 60-80 dungeons and raids. I find that 90% of the problems come from three reasons:

1. The npcbot cannot clearn up the combat state when the  combat  is over,then causes problems with the replica BOSS mechanism;
2. The npcbot casting spell takes a lot of public spell cooling time of the owner;
3. Npcbots cannot be regarded as a player, which causes a lot of spells and damage to be borne by the owner.

so, I suggest you to improve from these three aspects next. Thank you!

**Comment by trickerer** [comment]
1. Yes this may happen in dungeons sometimes, it's easilly remedied with `.npcbot recall tele` command
2. This is a bad translation and I'm not sure what you're trying to say
3. There was an attempt to allow targeting bots with any player-only spell but it was causing strange issues including crashes so it was disabled

**Reply by zbhcn** [reply]
If npcbots could be regarded as a player by boss Or in the dungeon，Players will have a better sense of experience

**Comment by zbhcn** [comment]
Thank you for your reply. I hope your code is getting better and better.
